diff -ur cx3110x-module-src.orig/src/sm_drv_spi.c cx3110x-module-src-2.0.15/src/sm_drv_spi.c
--- cx3110x-module-src.orig/src/sm_drv_spi.c	2008-09-16 01:37:49.000000000 +0400
+++ cx3110x-module-src-2.0.15/src/sm_drv_spi.c	2009-03-25 19:16:18.000000000 +0300
@@ -36,7 +36,6 @@
 #include <linux/platform_device.h>
 #include <linux/string.h>
 #include <linux/firmware.h>
-#include <linux/config.h>
 #if !defined(CONFIG_FW_LOADER) && !defined(CONFIG_FW_LOADER_MODULE)
 #error No Firmware Loading configured in the kernel !
 #endif
